### ISSUE TITLE
Language Models Are Unsupervised Multitask Learners

### ISSUE THREAD

**Issue by shnakazawa** [TeamAidemy/ds-paper-summaries#10]
Radford, Alec, Jeff Wu, Rewon Child, D. Luan, Dario Amodei, and Ilya Sutskever. 2019. “Language Models Are Unsupervised Multitask Learners.” https://www.semanticscholar.org/paper/9405cc0d6169988371b2755e573cc28650d14dfe.

- 2019年にOpenAIのチームが発表した仕事、GPT-2
- Unsupervised/Zero-shotは「タスクの明示的な教師あり学習を行わない」の意味
- Transformer + Attention + 大規模データ + 大規模モデル
- サイズの力で精度が改良できた + さらに改良できそうだという感触が得られた → さらに大規模なGPT-3へと繋がる

## Abstract
>Natural language processing tasks, such as question answering, machine translation, reading comprehension, and summarization, are typically approached with supervised learning on taskspecific datasets. We demonstrate that language models begin to learn these tasks without any explicit supervision when trained on a new dataset of millions of webpages called WebText. When conditioned on a document plus questions, the answers generated by the language model reach 55 F1 on the CoQA dataset matching or exceeding the performance of 3 out of 4 baseline systems without using the 127,000+ training examples. The capacity of the language model is essential to the success of zero-shot task transfer and increasing it improves performance in a log-linear fashion across tasks. Our largest model, GPT-2, is a 1.5B parameter Transformer that achieves state of the art results on 7 out of 8 tested language modeling datasets in a zero-shot setting but still underfits WebText. Samples from the model reflect these improvements and contain coherent paragraphs of text. These findings suggest a promising path towards building language processing systems which learn to perform tasks from their naturally occurring demonstrations

(DeepL翻訳)

質問応答、機械翻訳、読解、要約などの自然言語処理タスクは、通常、タスク固有のデータセットに対する教師あり学習でアプローチされる。我々は、WebTextと呼ばれる数百万のウェブページからなる新しいデータセットで学習した場合、言語モデルが明示的な教師なしでこれらのタスクを学習し始めることを実証する。文書と質問を条件とした場合、言語モデルによって生成された回答はCoQAデータセットで55F1に達し、127,000以上の学習例を用いないベースラインシステムの4つのうち3つの性能と同等かそれ以上である。言語モデルの容量は、ゼロショットタスク転送の成功に不可欠であり、これを増やすと、タスク間で対数線形的に性能が向上します。我々の最大のモデルであるGPT-2は、1.5BパラメータのTransformerで、ゼロショット設定においてテストされた8つの言語モデリングデータセットのうち7つで最先端の結果を達成しましたが、それでもWebTextには及びません。このモデルのサンプルは、これらの改善を反映し、首尾一貫した段落のテキストを含んでいます。これらの結果は、自然に発生するデモからタスクの実行を学習する言語処理システムの構築に向けた有望な道筋を示唆するものです。

## コード

https://github.com/openai/gpt-2

## 解決した課題/先行研究との比較

- これまでの言語モデルは用いるデータセットやタスクによってその性能が大きく左右されていた。
	- 「スペシャリスト」的モデルが中心。理由の一つとして、用いることのできるデータセットのサイズ制限ゆえ偏ったデータセットの方が扱いやすかったため。
- 「ゼネラリスト」を志向するモデルの研究も進められてきており、近年、Transformer (Vaswani et al., 2017) とAttentionを用いたモデルが大きな進歩をもたらした (GPT-1: Radford et al., 2018; BERT: Devlin et al., 2018)
- しかし、まだfine-tuningは必要。もっと減らせないか？
- 本論文はTransformer＋大量のデータセットを用いることで、ゼロショットで言語タスクを実行できることを示した。
	- (一部のデータではfine-tuningを行っているっぽい？本論文中では記述が見つからずだが、GPT-3論文で以下の記載有り)
	- >While [RWC+19] describe their work as “zero-shot task transfer” they sometimes provide examples of the relevant task in the context.

## 技術・手法のポイント

- 言語モデルを使用
- ほぼGPTと同じ。違いは
	- Layer normalization (Ba et al., 2016) の位置
	- Residual network (He et al., 2016) の位置
	- 巨大なデータサイズ、モデルサイズ
- 48層のdecoder block (15億パラメータ！)
- WebTextというデータセットを作成、使用 (800万サイト、4500万リンク、40GB)
	- 最初は Common Crawl によるWebをクローリングしたデータを使ったが、質の低い文章も多く含まれていた
	- そこで、Reddit で 3 karma 以上が付いた投稿に含まれるリンク先の文章のみを使ったデータセット「WebText」を作成
		- 一定数の人間により、良い記事と判断された文章のみを採用

## 評価指標
良いスコア (SOTA含)
- Language Model
- Children's Book test
- LAMBADA
- Winogrard Schema Challenge

課題が残る
- Reading Comprehension (CoQA)
- Summarization (ROUGE F1)

全然だめ
- Translation (WMT-14. 英-仏)
	- ただし、仏語のデータがものすごく少なかった (10 MB) ので、データ数が増えれば状況は変わりそう
- Question Answering (一問一答 from Natural Questions dataset (Kwiatkowski et al., 2019)
	- 4%. GPT-1では1%だった。データを増やし言語モデルを大きくすれば、精度が上がりそう

## 残された課題・議論
- 十分な精度が出ていないタスクもある。しかし、データ数を増やせば精度が上がりそうな気配がある
	- データを増やした方向で発展させたものがGPT-3

## 重要な引用
- Vaswani, Ashish, Noam Shazeer, Niki Parmar, Jakob Uszkoreit, Llion Jones, Aidan N. Gomez, Lukasz Kaiser, and Illia Polosukhin. 2017. “Attention Is All You Need.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1706.03762.
	- Transformer
- Radford, Alec, Karthik Narasimhan, Tim Salimans, and Ilya Sutskever. 2018. “Improving Language Understanding by Generative Pre-Training.” https://www.cs.ubc.ca/~amuham01/LING530/papers/radford2018improving.pdf.
	- GPT-1
- Devlin, Jacob, Ming-Wei Chang, Kenton Lee, and Kristina Toutanova. 2018. “BERT: Pre-Training of Deep Bidirectional Transformers for Language Understanding.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1810.04805.
	- BERT
- Wang, Alex, Amanpreet Singh, Julian Michael, Felix Hill, Omer Levy, and Samuel Bowman. 2018. “GLUE: A Multi-Task Benchmark and Analysis Platform for Natural Language Understanding.” In Proceedings of the 2018 EMNLP Workshop BLackboxNLP: Analyzing and Interpreting Neural Networks for NLP, 353–55. Brussels, Belgium: Association for Computational Linguistics.
	- GLUE
	- 「ゼネラリスト」的モデルの評価指標
- McCann, Bryan, Nitish Shirish Keskar, Caiming Xiong, and Richard Socher. 2018. “The Natural Language Decathlon: Multitask Learning as Question Answering.” arXiv [cs.CL]. arXiv. http://arxiv.org/abs/1806.08730.
	- decaNLP
	- 「ゼネラリスト」的モデルの評価指標

## 関係論文
- Brown, Tom B., Benjamin Mann, Nick Ryder, Melanie Subbiah, Jared Kaplan, Prafulla Dhariwal, Arvind Neelakantan, et al. 2020. “Language Models Are Few-Shot Learners.” 34th Conference on Neural Information Processing Systems. https://papers.nips.cc/paper/2020/hash/1457c0d6bfcb4967418bfb8ac142f64a-Abstract.html.
	- GPT-3

- 引用されていないが、Attention機構の初出 Bahdanau et al., 2014; Luong et al., 2015
